src/lib.rs

```diff
@@ -30,41 +33,18 @@ impl<T> NonEmpty<T> {
     }
 
     /// Create a new non-empty list with an initial element.
-    pub const fn singleton(e: T) -> Self {
-        NonEmpty(e, Vec::new())
+    pub const fn singleton(head: T) -> Self {
+        NonEmpty {
+            head,
+            tail: Vec::new(),
+        }
     }
 
     /// Always returns false.
     pub const fn is_empty(&self) -> bool {
         false
     }
 
-    /// Get the first element. Never fails.
-    pub const fn first(&self) -> &T {
-        &self.0
-    }
```

Comment:
I would prefer to keep these functions even if they aren't needed, to match the `Vec` API.

Reply:
Sure :ok_hand:

Reply:
Added it back :)